rust/rope/src/delta.rs

```diff
@@ -245,24 +246,43 @@ impl<N: NodeInfo> Delta<N> {
     }
 }
 
-impl<N: NodeInfo> fmt::Debug for Delta<N> {
+impl<N: NodeInfo> fmt::Debug for Delta<N> where Node<N>: fmt::Debug {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        try!(write!(f, "Delta("));
-        for el in &self.els {
-            match *el {
-                DeltaElement::Copy(beg,end) => {
-                    try!(write!(f, "[{},{}) ", beg, end));
+        if f.alternate() {
+            for el in &self.els {
+                match *el {
+                    DeltaElement::Copy(beg,end) => {
+                        try!(write!(f, "{}", "-".repeat(end-beg)));
```

Comment:
I think this is copy-pasted from code I wrote a while ago, but that was also after `?` was introduced. The real reason is I'm still getting used to remembering `?`, since most of my Rust experience before Google was before `?`.